tests/integration/helpers.py

```diff
@@ -179,24 +185,40 @@ async def run_mongo_op(
 
     output.succeeded = True
     if expecting_output:
-        try:
-            output.data = json.loads(stdout)
-        except Exception:
-            logger.error(
-                "Could not serialize the output into json.{}{}".format(
-                    f"\n\tSTDOUT:\n\t {stdout}" if stdout else "",
-                    f"\n\tSTDERR:\n\t {stderr}" if stderr else "",
-                )
-            )
-            logger.error(f"Failed to serialize output: {output}".format(output=stdout))
-            if not ignore_errors:
-                raise
-            else:
-                output.data = stdout
+        output.data = _process_mongo_operation_result(stdout, stderr, expect_json_load)
     logger.info("Done: '%s'", output)
     return output
 
 
+def _process_mongo_operation_result(stdout, stderr, expect_json_load):
```

Comment:
question - why was this needed?

Reply:
it is used in `run_mongo_op` to simplify the `run_mongo_op` function by splitting it in 2 parts otherwise lint is complaining